crypto/x509/by_dir.c

```diff
@@ -94,23 +92,28 @@ static int dir_ctrl(X509_LOOKUP *ctx, int cmd, const char *argp, long argl,
 
 static int new_dir(X509_LOOKUP *lu)
 {
-    BY_DIR *a;
+    BY_DIR *a = OPENSSL_malloc(sizeof(*a));
+
+    if (a == NULL)
+        goto err;
```

Comment:
I'd prefer to have three `X509err(X509_F_NEW_DIR, ERR_R_MALLOC_FAILURE)` calls in line 98 and line 101 and 107 (instead of goto's), to make the error cases distinguishable by their line number.